internal/remoteconfig/remoteconfig.go

```diff
@@ -0,0 +1,294 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package remoteconfig
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"net/http"
+	"os"
+	"time"
+
+	rc "github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/version"
+)
+
+// Callback represents a function that can process a remote config update.
+// A Callback function can be registered to a remote config client to automatically
+// react upon receiving updates.
+type Callback func(*rc.Update)
+
+// Capability represents a bit index to be set in clientData.Capabilites in order to register a client
+// for a specific capability
+type Capability uint
+
+const (
+	_ Capability = iota
+	// ASMActivation represents the capability to activate ASM through remote configuration
+	ASMActivation
+	// ASMIPBlocking represents the capability for ASM to block requests based on user IP
+	ASMIPBlocking
+	// ASMDDRules represents the capability to update the rules used by the ASM WAF for threat detection
+	ASMDDRules
+)
+
+// ClientConfig contains the required values to configure a remoteconfig client
+type ClientConfig struct {
+	// The address at which the agent is listening for remoteconfig update requests on
+	AgentAddr string
+	// The smenatic version of the user's application
+	AppVersion string
+	// The env this tracer is running in
+	Env string
+	// The rate at which the client should poll the agent for updates
```

Comment:
Isn't this technically an interval not a rate?

Reply:
Nice catch, rewording.